runtime/server_http_request_test.go

```diff
@@ -737,3 +738,42 @@ func TestGetQueryValues(t *testing.T) {
 	assert.Equal(t, []string(nil), lastQueryParam)
 	assert.Equal(t, 0, len(lastQueryParam))
 }
+
+func TestPeekBody(t *testing.T) {
+	gateway, err := benchGateway.CreateGateway(
+		defaultTestConfig,
+		defaultTestOptions,
+		exampleGateway.CreateGateway,
+	)
+
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer gateway.Close()
+
+	bgateway := gateway.(*benchGateway.BenchGateway)
+	bgateway.ActualGateway.HTTPRouter.Register(
+		"GET", "/foo", zanzibar.NewRouterEndpoint(
+			bgateway.ActualGateway.Logger,
+			bgateway.ActualGateway.AllHostScope,
+			"foo", "foo",
+			func(
+				ctx context.Context,
+				req *zanzibar.ServerHTTPRequest,
+				res *zanzibar.ServerHTTPResponse,
+			) {
+				req.RawBody = []byte(`{"arg1":{"b1":{"c1":"result"}}}`)
```

Comment:
this is a cheat test. You must call `ReadAll()` before we call `PeekBody()`; `ReadAll()` must become idempotent.